Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -168,6 +220,175 @@ namespace PhysX
+        {
+            return
+                letter == ':' || letter == '"' || letter == '\'' ||
+                letter == '{' || letter == '}' ||
+                letter == '<' || letter == '>';
+        };
+        AZStd::replace_if(m_bakedHeightfieldRelativePath.begin(), m_bakedHeightfieldRelativePath.end(), invalidCharacters, '_');
+        m_bakedHeightfieldRelativePath += AZ_FILESYSTEM_EXTENSION_SEPARATOR;
+        m_bakedHeightfieldRelativePath += Pipeline::HeightFieldAssetHandler::s_assetFileExtension;
+
+        {
+            AZ::Data::AssetId generatedAssetId;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(generatedAssetId,
+                &AZ::Data::AssetCatalogRequests::GenerateAssetIdTEMP, m_bakedHeightfieldRelativePath.c_str());
+
+            AZ::Data::Asset<Pipeline::HeightFieldAsset> asset = AZ::Data::AssetManager::Instance().FindAsset(generatedAssetId, AZ::Data::AssetLoadBehavior::Default);
+            if (!asset.GetId().IsValid())
+            {
+                asset = AZ::Data::AssetManager::Instance().CreateAsset<PhysX::Pipeline::HeightFieldAsset>(generatedAssetId);
+            }
+
+            m_bakedHeightfieldAsset = asset;
+
+            physx::PxHeightField* pxHeightfield = static_cast<physx::PxHeightField*>(m_shapeConfig->GetCachedNativeHeightfield());
+
+            // Since PxHeightfield will have shared ownership in both HeightfieldAsset and HeightfieldShapeConfiguration,
+            // we need to increment the reference counter here. Both of these places call release() in destructors,
+            // so we need to avoid double deletion this way.
+            pxHeightfield->acquireReference();
+
+            m_bakedHeightfieldAsset.Get()->SetHeightField(pxHeightfield);
+            m_bakedHeightfieldAsset.Get()->SetMinHeight(m_shapeConfig->GetMinHeightBounds());
+            m_bakedHeightfieldAsset.Get()->SetMaxHeight(m_shapeConfig->GetMaxHeightBounds());
+        }
+    }
+
+    bool EditorHeightfieldColliderComponent::CheckoutHeightfieldAsset() const
+    {
+        char projectPath[AZ_MAX_PATH_LEN];
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", projectPath, AZ_MAX_PATH_LEN);
+
+        AZStd::string heightfieldFullPath;
+        AzFramework::StringFunc::Path::ConstructFull(projectPath, m_bakedHeightfieldRelativePath.c_str(), heightfieldFullPath, true);
+
+        // make sure the folder is created
+        AZStd::string heightfieldCaptureFolderPath;
+        AzFramework::StringFunc::Path::GetFolderPath(heightfieldFullPath.data(), heightfieldCaptureFolderPath);
+        AZ::IO::SystemFile::CreateDir(heightfieldCaptureFolderPath.c_str());
+
+        // check out the file in source control                
+        bool checkedOutSuccessfully = false;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            checkedOutSuccessfully,
+            &AzToolsFramework::ToolsApplicationRequestBus::Events::RequestEditForFileBlocking,
```

Comment:
nit: I'm aware that it's convenient to use the blocking call, but it could leave the editor hanging depending on how the operation goes. An improvement would be to use the async call.